### ISSUE TITLE
sql: aggregation in window does not use index automatically

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#30586]
Reported by @bladefist on Gitter.
Forked from  #24415.

**BUG REPORT**

Please describe the issue you observed, and any steps we can take to reproduce it:

- Which version of CockroachDB are you using?

2.1 beta / master

- What did you do?

```sql
CREATE TABLE data (
  id UUID NULL,
  value INT NULL,
  col1 INT NULL,   
  col2 INT NULL,   
  col3 INT NULL,   
  col4 INT NULL,   
  col5 INT NULL,   
  col6 INT NULL,   
  col7 INT NULL,   
  col8 INT NULL,   
  col9 INT NULL,   
  col10 INT NULL,   
  INDEX foo (id ASC) STORING (value)
);

SELECT id, rank() over(order by sum(value) desc) FROM data GROUP BY id;
```

- What did you expect to see?

Query uses index foo

- What did you see instead?

Uses full table scan

- What was the impact?

Performance

cc @andy-kimball @RaduBerinde

**Comment by andy-kimball** [comment]
@justinj, can you check whether this will be fixed with your latest work on window functions?

**Comment by andy-kimball** [comment]
I just verified that this appears to be fixed in the latest master branch for the 19.2 release. The cost-based optimizer now handles window functions, and does the correct thing for this case.
